deps/src/caller.cpp

```diff
@@ -0,0 +1,391 @@
+    rChangeCurrRing(origin);
+    return reinterpret_cast<void *>(temp);
+}
+
+void singular_define_caller(jlcxx::Module & Singular)
+{
+    Singular.method("load_library", [](std::string name) {
+        char * plib = iiConvName(name.c_str());
+        idhdl  h = ggetid(plib);
+        omFree(plib);
+        if (h == NULL) {
+            BOOLEAN bo = iiLibCmd(omStrDup(name.c_str()), TRUE, TRUE, FALSE);
+            if (bo)
+                return jl_false;
+        }
+        return jl_true;
+    });
+    Singular.method("call_singular_library_procedure",
+                    &call_singular_library_procedure);
+    Singular.method("call_singular_library_procedure",
+                    &call_singular_library_procedure_wo_ring);
+    Singular.method("get_type_mapper", &get_type_mapper);
+    Singular.method("initialize_jl_c_types", &initialize_jl_c_types);
+
+
+    Singular.method("NUMBER_CMD_CASTER",
+                    [](void * obj) { return reinterpret_cast<number>(obj); });
+    Singular.method("RING_CMD_CASTER",
+                    [](void * obj) { return reinterpret_cast<ring>(obj); });
+    Singular.method("POLY_CMD_CASTER",
+                    [](void * obj) { return reinterpret_cast<poly>(obj); });
+    Singular.method("IDEAL_CMD_CASTER",
+                    [](void * obj) { return reinterpret_cast<ideal>(obj); });
+    Singular.method("INT_CMD_CASTER", [](void * obj) {
+        return jl_box_int64(reinterpret_cast<long>(obj));
+    });
+    Singular.method("STRING_CMD_CASTER", [](void * obj) {
+        return std::string(reinterpret_cast<char *>(obj));
+    });
+    Singular.method("INTVEC_CMD_CASTER", [](void * obj) {
+        return intvec_to_jl_array(reinterpret_cast<intvec *>(obj));
+    });
+    Singular.method("INTMAT_CMD_CASTER", [](void * obj) {
+        return intmat_to_jl_array(reinterpret_cast<intvec *>(obj));
+    });
+    Singular.method("BIGINT_CMD_CASTER", [](void * obj) {
+        return reinterpret_cast<__mpz_struct *>(obj);
+    });
+    Singular.method("BIGINTMAT_CMD_CASTER", [](void * obj) {
+        return reinterpret_cast<bigintmat *>(obj);
+    });
+    Singular.method("MAP_CMD_CASTER", [](void * obj) {
+        return reinterpret_cast<sip_smap *>(obj);
+    });
+    Singular.method("RESOLUTION_CMD_CASTER", [](void * obj) {
+        return reinterpret_cast<syStrategy>(obj);
+    });
+    Singular.method("LIST_CMD_TRAVERSAL", &convert_nested_list);
+    Singular.method("get_ring_content", &get_ring_content);
+    Singular.method("get_ring_ref", &get_ring_ref);
+    Singular.method("get_poly_ptr", &get_poly_ptr);
+    Singular.method("get_ideal_ptr", &get_ideal_ptr);
+    Singular.method("jl_array_to_intvec", &jl_array_to_intvec);
+    Singular.method("jl_array_to_intmat", &jl_array_to_intmat);
+    Singular.method("safe_singular_string", &safe_singular_string);
+
+    Singular.method("get_length_of_resolution", &get_length_of_resolution);
+    Singular.method("get_resolution_data", &get_resolution_data);
```

Comment:
The above two functions are never called. Intentional? Perhaps at least a test case should call them?

Reply:
No, it is also a leftover. Will remove it.